modules/auxiliary/gather/query_ldap.rb

```diff
@@ -0,0 +1,279 @@
+
+          Alternatively one can run one of several predefined queries by setting ACTION to the
+          appropriate value.
+
+          All results will be returned to the user in table format, with || as the delimiter
+          separating multiple items within one column.
+        },
+        'Author' => [
+          'Grant Willcox', # Module
+        ],
+        'References' => [
+        ],
+        'DisclosureDate' => '2022-05-19',
+        'License' => MSF_LICENSE,
+        'Actions' => actions,
+        'DefaultAction' => 'ENUM_ALL_OBJECTCLASS',
+        'DefaultOptions' => {
+          'SSL' => false
+        },
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'SideEffects' => [IOC_IN_LOGS],
+          'Reliability' => []
+        }
+      )
+    )
+
+    register_options([
+      Opt::RPORT(389), # Set to 636 for SSL/TLS
+      OptEnum.new('OUTPUT_FORMAT', [true, 'The output format to use', 'table', ['csv', 'table', 'json']]),
+      OptString.new('BASE_DN', [false, 'LDAP base DN if you already have it']),
+      OptString.new('QUERY_FILE_PATH', [false, 'Path to the JSON or YAML file to load and run queries from'], conditions: %w[ACTION == RUN_QUERY_FILE])
+    ])
+  end
+
+  def perform_ldap_query(ldap, filter, attributes)
+    returned_entries = ldap.search(base: @base_dn, filter: filter, attributes: attributes)
+    query_result = ldap.as_json['result']['ldap_result']
+    case query_result['resultCode']
+    when 0
+      vprint_good('Successfully queried LDAP server!')
+    when 1
+      print_error("Could not perform query #{filter}. Its likely the query requires authentication.")
+      print_error(query_result['errorMessage'])
+    else
+      print_error("Query #{filter} failed with error: #{query_result['errorMessage']}")
+    end
+    if returned_entries.nil? || returned_entries.empty?
+      print_error("No results found for #{filter}.")
+      nil
+    else
+      returned_entries
+    end
+  end
+
+  def generate_rex_tables(entries, format)
+    entries.each do |entry|
+      tbl = Rex::Text::Table.new(
+        'Header' => entry['dn'][0].split(',').join(' '),
+        'Indent' => 1,
+        'Columns' => ['Name', 'Attributes']
+      )
+
+      for attr in entry.attribute_names
+        if format == 'table'
+          tbl << [attr, entry[attr].join(' || ')] unless attr == :dn # Skip over DN entries for tables since DN information is shown in header.
+        else
+          tbl << [attr, entry[attr].join(' || ')] # DN information is not shown in CSV output as a header so keep DN entries in.
+        end
+      end
+
+      case format
+      when 'table'
+        print_status(tbl.to_s)
+      when 'csv'
+        print_status(tbl.to_csv)
+      else
+        print_error("Invalid format #{format} passed to generate_rex_tables!")
+        break
+      end
+    end
+  end
+
+  def output_json_data(entries, _columns)
```

Comment:
```suggestion
  def output_json_data(entries)
```

Reply:
Fixed in https://github.com/rapid7/metasploit-framework/pull/16598/commits/cca391757323058c5b6f8fcebe1e178ce21d3f97